GoogleChromeLabs/ps-analysis-tool
# Enhancement: Add filters in full report file

## Description
This PR updates callbacks handling data passing to report.html, updates the `useGlobalFiltering` hook handling filters, and passes filter values to the `useFiltering` hook.


## Relevant Technical Choices
- Pass the selected filters object to the `appliedFilters` key inside the PSAT_DATA object.
- The `useGlobalFiltering` hook will parse filter values from `globalThis.PSAT_DATA.appliedFilters` and select the ones present.


## Testing Instructions
- Open Extension, apply filters, and download the report from the landing page.
- The filters should be present in the downloaded report. 
- Now update the filters and download the report again.
- The zip file should be downloaded and report.html should have updated filters applied.
- Now try same on sitemap and single URL inside CLI.


## Additional Information:
Few minor issues found not related to the issue
- Added loader for the report download button.
- Memoize the clearQuery callback for fewer rerenders.
- Update keys for exemption cookies and exemption reasons translations.
- Landing page scrolling not updating the menu bar.


## Screenshot/Screencast
<img width="1470" alt="Screenshot 2024-07-23 at 17 23 37" src="https://github.com/user-attachments/assets/2a9ab87c-b053-44a4-8dfa-b518d103665d">



---

## Checklist



- ~This code is covered by unit tests to verify that it works as intended.~



Fixes #
